src/eth.rs

```diff
@@ -232,6 +238,17 @@ where
 
         Ok((mac, phy))
     }
+
+    fn eth_mac_rmii_config() -> eth_mac_config_t {
+        let mut config = Self::eth_mac_default_config();
+
+        #[cfg(any(esp_idf_version = "4.4", esp_idf_version_major = "5"))]
```

Comment:
1. Switch that to `#[cfg(not(esp_idf_version = "4.3"))]`. This way when version 6 comes out in 10 years, we are covered. :)
2. What does that mean for V4.3.2 (the minimum supported ESP-IDF version by us)? No clock configuration necessary? Also, what does that mean for `esp-idf-hal`? Is the `clock_config_t` even available on ESP-IDF V4.3.2?

Reply:
I have no idea :)

I copied that guard from https://github.com/esp-rs/esp-idf-svc/blob/master/src/eth.rs#L843

Reply:
Could it be that on V4.3.2 the only pin that is usable is gpio0?

Reply:
no, it seems to be the same in 4.3.2 as in 4.4: https://docs.espressif.com/projects/esp-idf/en/v4.3.2/esp32/api-reference/kconfig.html#config-eth-rmii-clk-mode

Reply:
So I really don't know why clock_config is not set on 4.3.2? Maybe it had a different format?

Reply:
So in 4.3 it was probably set up differently:

* 4.3 config: https://github.com/espressif/esp-idf/blob/release/v4.3/components/esp_eth/include/esp_eth_mac.h
* 4.4 config: https://github.com/espressif/esp-idf/blob/release/v4.4/components/esp_eth/include/esp_eth_mac.h

Reply:
I changed it to your suggestion